numalogic/transforms/_stateless.py

```diff
@@ -55,6 +61,84 @@ def inverse_transform(self, X) -> npt.NDArray[float]:
         return np.power(X, 1.0 / self.n) - self.add_factor
 
 
+class DataClipper(StatelessTransformer):
+    """
+    Applies column-wise ceiling transformation.
+
+    Args:
+    ----
+        lower: lower bound for clipping.
+        upper: upper bound for clipping.
+    """
+
+    __slots__ = ("lower", "upper")
+
+    def __init__(
+        self,
+        lower: Optional[Union[float, Sequence[float]]] = None,
+        upper: Optional[Union[float, Sequence[float]]] = None,
+    ):
+        self._validate_args(lower, upper)
+        self.lower = lower
+        self.upper = upper
+
+    @staticmethod
+    def _validate_args(
+        lower: Union[float, Sequence[float]], upper: Union[float, Sequence[float]]
+    ) -> None:
+        if lower is None and upper is None:
+            raise ValueError("At least one of lower or upper should be provided.")
+
+        if isinstance(lower, Sequence) and isinstance(upper, Sequence) and len(lower) != len(upper):
+            raise ValueError("lower and upper should have the same length.")
+
+    def transform(self, x: npt.NDArray[float], **__) -> npt.NDArray[float]:
+        _df = pd.DataFrame(x, dtype=np.float32)
+        if (self.lower is not None) and (self.upper is not None):
+            return _df.clip(lower=self.lower, upper=self.upper, axis=1).to_numpy(dtype=np.float32)
+        if self.upper is not None:
+            return _df.clip(upper=self.upper, axis=1).to_numpy(dtype=np.float32)
+        return _df.clip(lower=self.lower, axis=1).to_numpy(dtype=np.float32)
+
+
+class GaussianNoiseAdder(StatelessTransformer):
```

Comment:
This is probably a redundant comment. I don't find where this class is used, except in its unit test.

Reply:
Called from config